pages/ownerInformationPage.ts

```diff
@@ -79,9 +79,31 @@ export class OwnerInformationPage extends HelperBase{
     }
 
     async validatePetType(petName: string, petType: string){
-        const rosyPetType = this.page.getByText(petName).locator('..').locator('dd').last()
-        await expect(rosyPetType).toHaveText(petType)
+        const recievedPetType = this.page.getByText(petName).locator('..').locator('dd').last()
+        await expect(recievedPetType).toHaveText(petType)
 
     }
 
+    async validateOwnerInformationFromJSON(ownerObject: {}){
+
+        const ownerTable = this.page.getByRole('table').first()
+        await expect(ownerTable.locator('td').first()).toHaveText(`${ownerObject['firstName']} ${ownerObject['lastName']}`)
+        await expect(ownerTable.locator('td').nth(1)).toHaveText(ownerObject['address'])
+        await expect(ownerTable.locator('td').nth(2)).toHaveText(ownerObject['city'])
+        await expect(ownerTable.locator('td').last()).toHaveText(ownerObject['telephone'])
```

Comment:
I think this it's a little bit confusing implementation from a reusability perspective.
The method name is "validateOwnerInformationFromJSON". For the person who will read this method name for the first time, he will immediately have questions:
1. What information does this method validate?
2. What JSON should I use?
3. What should this JSON object's structure look like, and where do I get the data from?

Passing the parameters into this method individually will make it more readable and help you understand what values of owner information this method can validate.

Also, it seems like you forgot how to use user-visible locators :)
All locators in this method heavily rely on the index of the row in the table, which can be avoided.

Reply:
This makes so much sense!  I was debating which direction I should go.